frontend/src/MuiForms/MetricItem.tsx

```diff
@@ -0,0 +1,51 @@
+import CloseIcon from '@mui/icons-material/Close'
+import { IconButton, Stack, TextField, Tooltip } from '@mui/material'
+import { useTheme } from '@mui/material/styles'
+import _ from 'lodash-es'
+import { useCallback, useState } from 'react'
+import MessageAlert from 'src/MessageAlert'
+import { MetricValueWithId } from 'src/MuiForms/Metrics'
+import { isValidNumber } from 'utils/stringUtils'
+
+interface MetricItemProps {
+  metric: MetricValueWithId
+  onChange: (updatedMetric: MetricValueWithId) => void
+  onDelete: (metricId: string) => void
+}
+
+export default function MetricItem({ metric, onChange, onDelete }: MetricItemProps) {
+  const theme = useTheme()
+  const [errorMessage, setErrorMessage] = useState('')
+
+  const handleMetricValueOnChange = useCallback(
+    (value: string) => {
+      const valueAsStringOrNumber: string | number = value
+      parseInt(valueAsStringOrNumber)
+      setErrorMessage('')
+      if (!isValidNumber(value)) {
+        setErrorMessage('Metric value must be a valid whole number')
+      }
+      onChange({ id: metric.id, name: metric.name, value: valueAsStringOrNumber })
```

Comment:
This should be in an else block. Also with the current implementation `valueAsStringOrNumber` will never be a string.

Reply:
I've updated this, it effectively stops users from entering letters altogether, which might _not_ be a bad thing given the libraries existing constraints.